lib/sidekiq/grouping/redis.rb

```diff
@@ -141,7 +141,12 @@ def delete(name)
       end
 
       def remove_from_pending(name, batch_name)
-        redis { |conn| conn.zrem(pending_jobs(name), batch_name) }
+        redis do |conn|
+          conn.multi do |pipeline|
+            pipeline.zrem(pending_jobs(name), batch_name)
```

Comment:
Sorry, my brain isn't working full speed yet apparently. I was reading over this and realized something else. What do you think about reversing these two? That would guarantee that any interruption in the middle would be recoverable (since the supervisor would clean it up). That seems like it would be very unlikely to happen, but I guess theoretical reliability is what we're after here.

Reply:
That's a great idea!